applications: nrf_desktop: Forward boot report doc (#5590)
nrfconnect/sdk-nrf: applications/nrf_desktop/README.rst

For example, the mouse motion is forwarded as HID mouse report.

An nRF Desktop device supports the selected subset of the HID input reports.
For example, the nRF Desktop keyboard reference design (nrf52kbd_nrf52832) supports HID keyboard report, HID consumer control report and HID system control report.

> The following HID protocols are supported by nRF Desktop for HID input reports:

Reviewer: The HID protocol refers to HID reports in general, not only to HID input reports (e.g. we have HID boot keyboard LEDs report that is an HID output report)

Author: yes, but HID input reports can use only either boot or report protocol.

Reviewer: I would just suggest to separate HID report types (input, output and feature) from HID protocols (boot, report). I think that could be more understandable for the reader

Reviewer: Also under this section you currently mention data exchanged by device and host (and section refers only to HID input reports that come from device to the host)

* Report protocol - Most widely used in HID devices.

>    When establishing connection, HID device defines the device data in a HID descriptor and sends the data to the host.

Author: Just wanted to mark that it is this protocol with which most people would work with. Like this is the classic version and boot protocol is a special one.

* Boot protocol - Only available for mice and keyboards.
   No HID descriptor is used for this HID protocol.

>    Instead, fixed data packet formats must be used to send data from the HID device to the host.

Reviewer: are used to exchange data between HID device and host (keep in mind that we also have boot keyboard LEDs output report)

For example, the mouse motion is forwarded as HID mouse report.

An nRF Desktop device supports the selected subset of the HID input reports.
For example, the nRF Desktop keyboard reference design (nrf52kbd_nrf52832) supports HID keyboard report, HID consumer control report and HID system control report.

The following HID protocols are supported by nRF Desktop for HID input reports:

* Report protocol - Most widely used in HID devices.

>    When establishing connection, HID device sends a HID descriptor to the host.

Reviewer: I think it's more like host may read descriptor if host needs it

Reviewer: Still, it's not mandatory to read it on every reconnection

An nRF Desktop device supports the selected subset of the HID input reports.

> For example, the nRF Desktop keyboard reference desing (nrf52kbd_nrf52832) supports HID keyboard report, HID consumer control report and HID system control report.

Reviewer 2: ```suggestion
For example, the nRF Desktop keyboard reference design (nrf52kbd_nrf52832) supports HID keyboard report, HID consumer control report and HID system control report.
```

> HID input reports exist in severeal protocols. Following protocols are supported:

Reviewer 2: ```suggestion
The following HID protocols are supported by nRF Desktop for HID input reports:
```

> * report protocol - Most widely used in HID devices. When establishing connection, HID device defines their data in a HID descriptor and sends it to the host.

Reviewer 2: ```suggestion
* Report protocol - Most widely used in HID devices.
  When establishing connection, HID device defines the device data in a HID descriptor and sends the data to the host.
```

> * boot protocol - Only available for mouses and keyboards. No HID descriptor is used, insted fixed data packet formats must be used.

Reviewer 2: ```suggestion
* Boot protocol - Only available for mice and keyboards.
   No HID descriptor is used for this HID protocol.
   Instead, fixed data packet formats must be used to send data from the HID device to the host.
```
